src/main/java/core/basesyntax/serviceimpl/DataConverterImpl.java

```
@@ -0,0 +1,25 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (int i = 1; i < inputReport.size(); i++) {
+            String[] lineArray = inputReport.get(i).split(",");
+            String code = lineArray[0].trim();
+            String fruit = lineArray[1].trim();
+            int quantity = Integer.parseInt(lineArray[2].trim());
```

Ensure that any parsing/validation failures result in informative RuntimeExceptions (including cause when appropriate) instead of raw exceptions propagating. The checklist requires rethrowing informative RuntimeExceptions rather than printing stack traces. For quantity parsing and other checks (lines 13–18), wrap exceptions with clear messages and include the original exception as the cause. See checklist guidance: "Don't ignore exceptions in try-catch blocks ... rethrow a RuntimeException with an informative message and exception object."